src/SimplePie.php

```diff
@@ -1259,6 +1291,14 @@ public function set_source_class(string $class = Source::class)
      */
     public function set_useragent(?string $ua = null)
     {
+        if ($this->http_client_injected) {
+            throw new SimplePieException(sprintf(
+                'Using "%s()" has no effect, because you already provided a HTTP client with "%s::set_http_client()". Configure user agent string in your HTTP client instead.',
+                __METHOD__,
+                self::class
+            ));
+        }
+
```

Comment:
We should unset a possible created $this->http_client on calling `set_useragent()`.
```suggestion

        // Reset a possible existing FileClient,
        // so a new client with the changed value will be created
        $this->http_client = null;
```